R/helper_additional_functions.R

```diff
@@ -213,7 +218,33 @@ clean_pbp <- function(pbp, ...) {
         dplyr::vars(.data$passer_id, .data$rusher_id, .data$receiver_id, .data$id, ends_with("player_id")),
         update_ids, legacy_id_map) %>%
       dplyr::arrange(.data$index) %>%
-      dplyr::select(-"index")
+      dplyr::select(-"index") %>%
+      # add action player
+      dplyr::mutate(
+        fantasy_player_name = case_when(
+          !is.na(.data$rusher_player_name) ~ .data$rusher_player_name,
+          is.na(.data$rusher_player_name) & !is.na(.data$receiver_player_name) ~ .data$receiver_player_name,
+          TRUE ~ NA_character_
```

Comment:
Right we need to pick one of these:
* Targeted receiver on pass plays with target, rusher on run plays, NA otherwise
* Targeted receiver on pass plays with target, rusher on run plays, QB on pass plays with no target, QB on plays with sack

I think the latter is somewhat confusing- if you're looking at this column alone, you wouldn't be able to tell if a given player in the column was a receiver on a pass play or a quarterback throwing the ball away.

Reply:
Yeah I agree. I had a thought error because I assumed we were losing scramble yards but that's actually wrong because the QB will be listed as `rusher_player_name`. However, this is not true for the `fantasy` column because we define scramblers as `passer`